chore: Unify imports syntax in admin-e2e, add missing dependencies

tomalec opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Unify the import syntax to use ESM `import/export` over legacy CJS `require`. This not only makes the code easier to read and lets devs avoid switching between conventions, but also improves static analysis, therefore IDE annotations and linter. 

That helped to find missing dependencies, and also highlight another problem:
![image](https://user-images.githubusercontent.com/17435/179018705-3c2dc93d-4469-43ed-9108-18b1e2cedffa.png)

that may be related to the problems I face in #33852 

### How to test the changes in this Pull Request:

1. Run the build
2. Run tests

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

tomalec committed:
Unify test spec helpers `export` syntax to ESM,

to make it possible to impor them locally.

tomalec committed:
Unify import syntax to ESM

tomalec committed:
Install missing dependencies of admin-e2e-tests

tomalec commented:
Regarding, linting error and failing tests

TBH, I don't get how it was supposed to work before.
For example line https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/admin-e2e-tests/src/specs/activate-and-setup/basic-setup.ts#L65 uses `toMatchElement` which requires `jest-puppeteer` matcher, but we explicitly import/require original jest's one in https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/admin-e2e-tests/src/specs/activate-and-setup/basic-setup.ts#L18-L19

tomalec commented:
Marking as ready for review. In my opinion failing linters are a sign of an actual issue to be fixed, but not introduced here.

chihsuan commented:
Hi @tomalec Thanks for working on this! 

> Marking as ready for review. In my opinion failing linters are a sign of an actual issue to be fixed, but not introduced here.

Yea, I agree. To merge this, I'm wondering if you could perhaps use `@ts-ignore` or `@ts-expect-error` so that the GH actions could pass since it's important to ensure the TS is not broken. Thank you!

tomalec commented:
> I'm wondering if you could perhaps use `@ts-ignore` or `@ts-expect-error` so that the GH actions could pass since it's important to ensure the TS is not broken.

I can. But I'm, not sure if that's a good idea. Since it won't fix TS, it will still be broken but will ignore the lining error.

chihsuan commented:
> > I'm wondering if you could perhaps use `@ts-ignore` or `@ts-expect-error` so that the GH actions could pass since it's important to ensure the TS is not broken.
> 
> I can. But I'm, not sure if that's a good idea. Since it won't fix TS, it will still be broken but will ignore the lining error.

Okay, do you plan to fix the TS error in another PR?

tomalec commented:
> Okay, do you plan to fix the TS error in another PR?

I'd love to but, I don't know how, as TBH I have no idea why and how it was meant to work in the first place.

tomalec committed:
Do not import `expect` from `@jest/globals` in admin-e2e-tests

Let typescript pick the puppeteer extension from global.
Address https://github.com/woocommerce/woocommerce/pull/33901#issuecomment-1221650234

tomalec committed:
Map unknown types in `admin-e2e-tests` to make tsc happy.

tomalec committed:
Add changelog entry

tomalec commented:
I have no idea why PHPCS is failing, as I did;t touch any PHP file.

Speaking of [JS tests failing](https://github.com/woocommerce/woocommerce/runs/8019982493?check_suite_focus=true), I think it may be related to jest versions mismatch - fixed in https://github.com/woocommerce/woocommerce/pull/34322

chihsuan commented on packages/js/admin-e2e-tests/src/fixtures/http-client.ts:
```diff
@@ -2,13 +2,15 @@
 // This is necessary so that it can make requests to the REST API.
-const admin = config.get( 'users.admin' );
-const url = config.get( 'url' );
+const admin = config.get( 'users.admin' ) as {
+	username: string;
+	password: string;
+};
+const url = config.get( 'url' ) as string;
```
@tomalec Instead of using type assertion, I think it would be great if we could just use the angle-bracket syntax since it's native-supported by `config` module.

```ts
const admin = config.get< {
	username: string;
	password: string;
} >( 'users.admin' );
const url = config.get< string >( 'url' );
```

tomalec replied:
```suggestion
const admin = config.get< {
	username: string;
	password: string;
} >( 'users.admin' );
const url = config.get< string >( 'url' );
```

chihsuan commented on packages/js/admin-e2e-tests/src/pages/BasePage.ts:
```diff
@@ -11,10 +15,7 @@ import { DropdownTypeaheadField } from '../elements/DropdownTypeaheadField';
-const config = require( 'config' );
-/* eslint-enable @typescript-eslint/no-var-requires */
-const baseUrl = config.get( 'url' );
+const baseUrl = config.get( 'url' ) as string;
```
```suggestion
const baseUrl = config.get< string >( 'url' );
```

chihsuan commented:
> I have no idea why PHPCS is failing, as I did;t touch any PHP file.
> 
> Speaking of [JS tests failing](https://github.com/woocommerce/woocommerce/runs/8019982493?check_suite_focus=true), I think it may be related to jest versions mismatch - fixed in #34322

@tomalec Thanks for addressing the issue so quickly! I agree failing PHP action is unrelated to this PR.

tomalec committed:
Use the angle-bracket syntax instead of type assertions

Address https://github.com/woocommerce/woocommerce/pull/33901#discussion_r955591953

Co-authored-by: Chi-Hsuan Huang <chihsuan.tw@gmail.com>

tomalec commented:
@chihsuan I addressed your comments, would you mind another check?

chihsuan reviewed (approved):
@tomalec Thank you for working on this! Looks good to me. 👍
